hashicorp/boundary-ui
# Chore upgrade eslint and deps

## Description
Upgrade eslint and plugins:

- `eslint` to `8.54.0` [link to changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- `eslint-config-prettier` to `9.0.0` [link to changelog](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md)
- `eslint-plugin-ember` to `11.11.1` [link to changelog](https://github.com/ember-cli/eslint-plugin-ember/releases?page=1)
- `eslint-plugin-n` to `16.3.1` [link to changelog](https://github.com/eslint-community/eslint-plugin-n/releases)

Missing package upgrade: `eslint-plugin-qunit`. This will come in a separate PR because it has side effects.

After each upgrade I run `yarn lint && yarn format` without seeing changes. So the upgrade didn't bring new linting rules that are not compliance with our codebase.

## Screenshots:
Before the upgrade (run `yarn outdated`):
![Screenshot 2023-11-24 at 12 01 36 PM](https://github.com/hashicorp/boundary-ui/assets/9775006/d198d79d-a37e-45de-8e63-1c40e4b43d78)

After the upgrade (run `yarn outdated`):
![Screenshot 2023-11-24 at 11 59 15 AM](https://github.com/hashicorp/boundary-ui/assets/9775006/926daff2-0b74-40a2-8288-40c8e2627153)


## How to Test
- Within each addon and UI run: `yarn lint && yarn format` and see no changes.
- Run GH validate monorepo and see no changes/warnings.
- Run `boundary-ui-release` to see it does not bring any errors/warnings and the CI is happy with the changes. [Link to the successful run](https://github.com/hashicorp/boundary-ui-releases/actions/runs/6984530751).

## Checklist:


~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
